Update liblinphone and dependencies

jluttine opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Update liblinphone and its dependencies.

This PR should be favored over #116316, #115565, #109924 and #109882.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

r-ryantm committed:
soci: 4.0.0 -> 4.0.1

jluttine committed:
bctoolbox: 4.4.24 -> 4.4.34

jluttine committed:
ortp: 4.4.32 -> 4.4.34

jluttine committed:
mediastreamer: 4.4.32 -> 4.4.35

jluttine committed:
lime: 4.4.21 -> 4.4.34

jluttine committed:
belcard: 4.4.24 -> 4.4.34

jluttine committed:
belr: 4.3.2 -> 4.4.34

jluttine committed:
bzrtp: 4.4.9 -> 4.4.34

jluttine committed:
belle-sip: 4.4.26 -> 4.4.34

jluttine committed:
liblinphone: 4.4.21 -> 4.4.35

jluttine commented:
```
$ nixpkgs-review pr 116768
...
14 packages built:
bctoolbox belcard belle-sip belr bzrtp liblinphone lime linphone mediastreamer mediastreamer-openh264 msilbc ortp soci trx
```

jluttine commented:
Tested execution of the user-facing binary `linphone` and it worked.

SuperSandro2000 commented on pkgs/development/libraries/bzrtp/default.nix:
```diff
@@ -7,7 +7,7 @@
+  version = "4.4.34";
 
   src = fetchFromGitLab {
     domain = "gitlab.linphone.org";
```
You missed updating the hash.

jluttine replied:
Well spotted! But, actually, I didn't.. The tags just seem to point to the same commit. :open_mouth: 

https://gitlab.linphone.org/BC/public/bzrtp/-/tags

And it looks like there really hasn't been many commits even though the the tag has been bumped a lot:

https://gitlab.linphone.org/BC/public/bzrtp/-/commits/master/

So, I suppose they (Linphone) just tag all the projects so that their versions are in sync even if some projects haven't had any changes they want to release yet. Or that's my interpretation. Weird.

r-rmcgibbo commented:
Result of `nixpkgs-review pr 116768` at 5c1e77bd run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build (new failures):</summary>
  <ul>
    <li><a href="https://gist.github.com/a362535e4b174d4bfb68112503a49fcd">belle-sip</a></li></li>
    <li><a href="https://gist.github.com/15bf2ca9b297e8357887e146076fff7d">lime</a></li></li>
  </ul>
</details>
<details>
  <summary>2 packages skipped due to time constraints:</summary>
  <ul>
    <li>liblinphone</li>
    <li>linphone</li>
  </ul>
</details>
<details>
  <summary>10 packages built successfully:</summary>
  <ul>
    <li><a href="https://gist.github.com/cd17f9ab1f06eb8c10b94f70eaa7edb8">bctoolbox</a></li></li>
    <li><a href="https://gist.github.com/57d47974a8a56037bb78e17b2d45031f">belcard</a></li></li>
    <li><a href="https://gist.github.com/1913bf7a35f10430c3e42e993a222cf9">belr</a></li></li>
    <li><a href="https://gist.github.com/ac9848a747a6b13907b4676ba187da27">bzrtp</a></li></li>
    <li><a href="https://gist.github.com/92400e29663a0ff4683ca6e50bc0da7c">mediastreamer</a></li></li>
    <li><a href="https://gist.github.com/e902beeeef5af30239fba49552706df8">mediastreamer-openh264</a></li></li>
    <li><a href="https://gist.github.com/3dbb7e03393dec5cb5372474a1a259f7">msilbc</a></li></li>
    <li><a href="https://gist.github.com/621e068444c98c5829c15630ade8ddf1">ortp</a></li></li>
    <li><a href="https://gist.github.com/3135f80224d465bf4d192a679d142bb9">soci</a></li></li>
    <li><a href="https://gist.github.com/f73e2b00f17c8f3c5161933461216dba">trx</a></li></li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/libraries/lime/default.nix:32:5:
```
   |
32 |     license = licenses.gpl3;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/libraries/mediastreamer/default.nix:105:5:
```
    |
105 |     license = licenses.gpl3;
    |     ^
```
</li>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/development/libraries/mediastreamer/default.nix:7:3:
```
  |
7 | , fetchpatch
  |   ^
```
</li>
  </ul>
</details>

jluttine commented:
@SuperSandro2000 Ah, perhaps you meant to ask marking the package broken on aarch64 on *this* PR instead of https://github.com/NixOS/nixpkgs/pull/116780#issuecomment-802350061 ? :slightly_smiling_face:

jluttine committed:
libantlr3c: mark broken on aarch64

The package failed to build with error:

gcc: error: unrecognized command line option '-m64'

See:

https://gist.github.com/r-rmcgibbo/15bf2ca9b297e8357887e146076fff7d

https://gist.github.com/r-rmcgibbo/a362535e4b174d4bfb68112503a49fcd

jluttine committed:
belle-sip: apply nixpkgs-fmt

jluttine committed:
lime: apply nixpkgs-fmt

jluttine commented:
It seems that it was `libantlr3c` that failed to build on aarch64, so I marked only that package broken.

SuperSandro2000 commented:
> @SuperSandro2000 Ah, perhaps you meant to ask marking the package broken on aarch64 on _this_ PR instead of [#116780 (comment)](https://github.com/NixOS/nixpkgs/pull/116780#issuecomment-802350061) ? 🙂

I don't know. could be.

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 116768` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
14 packages built:
</summary>
<ul>
<li>bctoolbox</li>
<li>belcard</li>
<li>belle-sip</li>
<li>belr</li>
<li>bzrtp</li>
<li>liblinphone</li>
<li>lime</li>
<li>linphone</li>
<li>mediastreamer</li>
<li>mediastreamer-openh264</li>
<li>msilbc</li>
<li>ortp</li>
<li>soci</li>
<li>trx</li>
</ul>
</details>
